internal/profiles/validator.go

```diff
@@ -33,19 +33,13 @@ import (
 )
 
 // Validator encapsulates the logic for validating profiles
-type Validator struct {
-	store db.Store
-}
-
-// NewValidator is a factory method for the Validator struct
-func NewValidator(store db.Store) *Validator {
-	return &Validator{store}
-}
+type Validator struct{}
```

Comment:
I had to make `Validator` transaction-aware. The problem I encountered was that while creating the rule types and profile in the same transaction, the validator would not see the new rule types.

Reply:
Should we do that in a separate PR to keep this one small and targetted?